### PR TITLE
LuaJIT: bump, fix issue with `pcall` on `aarch64-darwin` 

### PR DESCRIPTION
**Pull request by lblasc**
###### Motivation for this change

**Make `aarch64-darwin` work with upstream luajit**

Actual issue is that `pcall` doesn't work when unwind external is enabled, for more details see upstream issue: https://github.com/LuaJIT/LuaJIT/issues/698
* patch Makefile to disable unwind external only on `aarch64-darwin`
* openresty luajit flavor is not needed (revert #134791 ) - I don't see reason for keeping it
* make neovim to use plain luajit

I don't have `aarch64-darwin` but I have good friends who have it :) thx @zuigon for testing:
```
λ nix-build -A luajit.pkgs.cjson
/nix/store/i9csarm458qm268qy6hzjy7iqyvqnpq1-luajit-2.1.0-2021-08-12-lua-cjson-2.1.0.6-1
λ nix-build -A luajit_2_1
/nix/store/fi6hramx8ajfic5snjn1g54ngxkcgahv-luajit-2.1.0-2021-08-12
λ ./result/bin/luajit -v
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2021 Mike Pall. https://luajit.org/
λ echo "pcall(require, 'nope'); print('ok')" | ./result/bin/luajit
ok
```

**Bump luajit 2.0 & 2.1**

*luajit_2_0*
https://github.com/LuaJIT/LuaJIT/compare/98f95f69180d48ce49289d6428b46a9ccdd67a46...3a654999c6f00de4cb9e61232d23579442e544a0

Nothing dangerous, just some fixes.

*luajit_2_1*
https://github.com/LuaJIT/LuaJIT/compare/e957737650e060d5bf1c2909b741cc3dffe073ac...8ff09d9f5ad5b037926be2a50dc32b681c5e7597

Fixes and more string buffer changes.

cc @gpanders @rvolosatovs @vcunat 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gpanders** [comment]
> openresty luajit flavor is not needed (revert Use luajit-openresty with Neovim on aarch64-darwin #134791 ) - I don't see reason for keeping it

No complaints here. As long as LuaJIT works I don't care which one we're using.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135672` at 13a1ac54 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages_hardkernel_4_14.bcc</li>
    <li>linuxPackages_hardkernel_4_14.bpftrace</li>
    <li>linuxPackages_hardkernel_4_14.sysdig</li>
    <li>linuxPackages_hardkernel_latest.bcc</li>
    <li>linuxPackages_hardkernel_latest.bpftrace</li>
    <li>linuxPackages_hardkernel_latest.sysdig</li>
  </ul>
</details>
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/feeab07e48feb070fd7071f85562e200">gimpPlugins.texturize</a></li>
    <li><a href="https://gist.github.com/c087a6dbccf854f057d54e050a65f7b4">glimpsePlugins.texturize</a></li>
    <li><a href="https://gist.github.com/8d02411e01c0e73978547b66c6d663c1">luajitPackages.digestif</a></li>
    <li><a href="https://gist.github.com/9a41276704aaae44bec1ddc9f007c6bb">luajitPackages.gitsigns-nvim</a></li>
    <li><a href="https://gist.github.com/b1d116b6d6d816ebd5ff8fac8e9bc222">luajitPackages.ltermbox</a></li>
    <li><a href="https://gist.github.com/f60d5831204f33d7b9edd3d467ca407a">luajitPackages.luaffi</a></li>
    <li><a href="https://gist.github.com/2b06feda2a48e50608b18ab157b049b6">luajitPackages.plenary-nvim</a></li>
  </ul>
</details>
<details>
  <summary>103 packages skipped due to time constraints:</summary>
  <ul>
    <li>_90secondportraits</li>
    <li>cgit</li>
    <li>enlightenment.econnman</li>
    <li>enlightenment.efl</li>
    <li>enlightenment.enlightenment</li>
    <li>enlightenment.ephoto</li>
    <li>enlightenment.evisum</li>
    <li>enlightenment.rage</li>
    <li>enlightenment.terminology</li>
    <li>gimp-with-plugins</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>138 packages built successfully:</summary>
  <ul>
    <li>arcan.arcan</li>
    <li>arcan.arcan-wrapped</li>
    <li>arcan.durden-wrapped</li>
    <li>arcan.everyone-wrapped</li>
    <li>arcan.pipeworld-wrapped</li>
    <li>arcan.prio-wrapped</li>
    <li>arcan.xarcan</li>
    <li>gegl</li>
    <li>gimp (gimpPlugins.gimp)</li>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.resynthesizer</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>glimpse (glimpsePlugins.gimp)</li>
    <li>glimpsePlugins.exposureBlend</li>
    <li>glimpsePlugins.farbfeld</li>
    <li>glimpsePlugins.fourier</li>
    <li>glimpsePlugins.gap</li>
    <li>glimpsePlugins.gimplensfun</li>
    <li>glimpsePlugins.lightning</li>
    <li>glimpsePlugins.lqrPlugin</li>
    <li>glimpsePlugins.resynthesizer</li>
    <li>glimpsePlugins.waveletSharpen</li>
    <li>gnvim</li>
    <li>knot-resolver</li>
    <li>koreader</li>
    <li>luajit (luajitPackages.lua ,luajit_2_1)</li>
    <li>luajitPackages.alt-getopt</li>
    <li>luajitPackages.ansicolors</li>
    <li>luajitPackages.argparse</li>
    <li>luajitPackages.basexx</li>
    <li>luajitPackages.binaryheap</li>
    <li>luajitPackages.bit32</li>
    <li>luajitPackages.busted</li>
    <li>luajitPackages.cassowary</li>
    <li>luajitPackages.cjson (luajitPackages.lua-cjson)</li>
    <li>luajitPackages.compat53</li>
    <li>luajitPackages.cosmo</li>
    <li>luajitPackages.coxpcall</li>
    <li>luajitPackages.cqueues</li>
    <li>luajitPackages.cyrussasl</li>
    <li>luajitPackages.dkjson</li>
    <li>luajitPackages.fifo</li>
    <li>luajitPackages.http</li>
    <li>luajitPackages.inspect</li>
    <li>luajitPackages.ldbus</li>
    <li>luajitPackages.ldoc</li>
    <li>luajitPackages.lgi</li>
    <li>luajitPackages.linenoise</li>
    <li>luajitPackages.ljsyscall</li>
    <li>luajitPackages.lpeg</li>
    <li>luajitPackages.lpeg_patterns</li>
    <li>luajitPackages.lpeglabel</li>
    <li>luajitPackages.lpty</li>
    <li>luajitPackages.lrexlib-gnu</li>
    <li>luajitPackages.lrexlib-pcre</li>
    <li>luajitPackages.lrexlib-posix</li>
    <li>luajitPackages.lua-cmsgpack</li>
    <li>luajitPackages.lua-iconv</li>
    <li>luajitPackages.lua-lsp</li>
    <li>luajitPackages.lua-messagepack</li>
    <li>luajitPackages.lua-resty-http</li>
    <li>luajitPackages.lua-resty-jwt</li>
    <li>luajitPackages.lua-resty-openidc</li>
    <li>luajitPackages.lua-resty-openssl</li>
    <li>luajitPackages.lua-resty-session</li>
    <li>luajitPackages.lua-term</li>
    <li>luajitPackages.lua-toml</li>
    <li>luajitPackages.lua-yajl</li>
    <li>luajitPackages.lua-zlib</li>
    <li>luajitPackages.lua_cliargs</li>
    <li>luajitPackages.luabitop</li>
    <li>luajitPackages.luacheck</li>
    <li>luajitPackages.luacov</li>
    <li>luajitPackages.luadbi</li>
    <li>luajitPackages.luadbi-mysql</li>
    <li>luajitPackages.luadbi-postgresql</li>
    <li>luajitPackages.luadbi-sqlite3</li>
    <li>luajitPackages.luadoc</li>
    <li>luajitPackages.luaepnf</li>
    <li>luajitPackages.luaevent</li>
    <li>luajitPackages.luaexpat</li>
    <li>luajitPackages.luafilesystem</li>
    <li>luajitPackages.lualogging</li>
    <li>luajitPackages.luaossl</li>
    <li>luajitPackages.luaposix</li>
    <li>luajitPackages.luarepl</li>
    <li>luajitPackages.luarocks</li>
    <li>luajitPackages.luarocks-nix</li>
    <li>luajitPackages.luasec</li>
    <li>luajitPackages.luasocket</li>
    <li>luajitPackages.luasql-sqlite3</li>
    <li>luajitPackages.luassert</li>
    <li>luajitPackages.luasystem</li>
    <li>luajitPackages.luautf8</li>
    <li>luajitPackages.luazip</li>
    <li>luajitPackages.luuid</li>
    <li>luajitPackages.luv</li>
    <li>luajitPackages.luxio</li>
    <li>luajitPackages.lyaml</li>
    <li>luajitPackages.markdown</li>
    <li>luajitPackages.mediator_lua</li>
    <li>luajitPackages.moonscript</li>
    <li>luajitPackages.mpack</li>
    <li>luajitPackages.nvim-client</li>
    <li>luajitPackages.penlight</li>
    <li>luajitPackages.rapidjson</li>
    <li>luajitPackages.readline</li>
    <li>luajitPackages.say</li>
    <li>luajitPackages.std-_debug</li>
    <li>luajitPackages.std-normalize</li>
    <li>luajitPackages.stdlib</li>
    <li>luajitPackages.vicious</li>
    <li>luajitPackages.vstruct</li>
    <li>luajitPackages.wrapLua</li>
    <li>luajit_2_0</li>
    <li>minetest (minetestclient_5)</li>
    <li>minetestserver_5</li>
    <li>neovim-qt</li>
    <li>neovim-qt-unwrapped</li>
    <li>neovim-remote</li>
    <li>neovim-unwrapped</li>
    <li>nvimpager</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-nvfbc</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>riko4</li>
    <li>wrk</li>
    <li>wrk2</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/interpreters/luajit/default.nix:45:10:
```
   |
45 |   inherit name version;
   |          ^
```

Near pkgs/development/interpreters/luajit/default.nix:45:10:
```
   |
45 |   inherit name version;
   |          ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/interpreters/luajit/default.nix [hunk 1]
```diff
@@ -26,28 +27,34 @@ assert enableJITDebugModule -> enableJIT;
...
+    owner = "LuaJIT";
+    repo = "LuaJIT";
+    inherit sha256 rev;
   };
 
   luaversion = "5.1";
 
+  # Fix for pcall on aarch64-darwin.
+  # Upstream issue: https://github.com/LuaJIT/LuaJIT/issues/698
+  patches = lib.optionals (stdenv.hostPlatform.system == "aarch64-darwin") [ ./aarch64-darwin-disable-unwind-external.patch ];
```
[the remark this answers is not in the document]
Comment (lblasc) [reply pkgs/development/interpreters/luajit/default.nix:56]: @SuperSandro2000 are you fine with this approach ? If yes, lets merge this one.

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/interpreters/luajit/default.nix:56]: Conditional patching is not fine when it is not absolutely necessary. The next person who bumps this package which might break the patch will most likely not check it on any aarch64 platform which will probably result in an outdated patch.

File: pkgs/development/interpreters/luajit/default.nix [hunk 2]
```diff
@@ -94,16 +103,16 @@ stdenv.mkDerivation rec {
...
   };
 
   meta = with lib; {
     description = "High-performance JIT compiler for Lua 5.1";
-    homepage    = "http://luajit.org";
-    license     = licenses.mit;
-    platforms   = platforms.linux ++ platforms.darwin;
+    homepage = "http://luajit.org";
+    license = licenses.mit;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/luajit/default.nix:115]: This could/should be unix.

Comment (vcunat) [reply pkgs/development/interpreters/luajit/default.nix:115]: Even `all` I suppose... at least the support upstream is very wide: https://luajit.org/luajit.html
But for Hydra this won't do any change, as we only have linux and darwin there (currently).

File: pkgs/development/interpreters/luajit/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (lblasc) [reply pkgs/development/interpreters/luajit/default.nix:56]: I agree, for now I think this is the most elegant way I can think of to only patch 2.1 version without braking 2.0.
In next update I will remove this patch, upstream included it https://github.com/LuaJIT/LuaJIT/commit/27ee3bcd79b12a0c71f00427ee1a2e486c684486

**Pull request merged**